test-d/hooks.ts

```diff
@@ -17,6 +17,14 @@ After(function () {})
 BeforeStep(function () {})
 AfterStep(function () {})
 
+// should allow hook functions to be async
+BeforeAll(async function () {})
```

Comment:
Did you not want to use our own types here for these functions?

e.g. for `BeforeStep` and `AfterStep` could we not use the `TestStepHookFunction` from above? Or am I missing something?

Reply:
That type is implicitly being used - the goal with this test is to write what a user would typically write and make sure it passes the type check.

Reply:
💡  thanks for clarifying!